cou/steps/plan.py

```diff
@@ -185,30 +185,30 @@ def verify_hypervisors_cli_input(args: CLIargs, analysis_result: Analysis) -> No
     :type analysis_result: Analysis
     """
     if args.is_hypervisors_command:
-        if cli_machines := args.machines:
-            verify_data_plane_cli_machines(cli_machines, analysis_result)
+        verify_data_plane_cli_machines(args.machines, analysis_result)
+        verify_data_plane_cli_azs(args.availability_zones, analysis_result)
 
-        elif cli_azs := args.availability_zones:
-            verify_data_plane_cli_azs(cli_azs, analysis_result)
 
-
-def verify_data_plane_cli_machines(cli_machines: set[str], analysis_result: Analysis) -> None:
+def verify_data_plane_cli_machines(
+    cli_machines: Optional[set[str]], analysis_result: Analysis
+) -> None:
     """Verify if the machines passed from the CLI are valid.
 
     :param cli_machines: Machines passed to the CLI as arguments
     :type cli_machines: set[str]
     :param analysis_result: Analysis result
     :type analysis_result: Analysis
     """
-    verify_data_plane_membership(
-        all_options=set(analysis_result.machines.keys()),
-        data_plane_options=set(analysis_result.data_plane_machines.keys()),
-        cli_input=cli_machines,
-        parameter_type="Machine(s)",
-    )
+    if cli_machines:
+        verify_data_plane_membership(
+            all_options=set(analysis_result.machines.keys()),
+            data_plane_options=set(analysis_result.data_plane_machines.keys()),
+            cli_input=cli_machines,
+            parameter_type="Machine(s)",
+        )
```

Comment:
not-blocking: here will not brink much benefit, but in next function it will make much more sense. 
```suggestion
    if not cli_machines:
        return

    verify_data_plane_membership(
        all_options=set(analysis_result.machines.keys()),
        data_plane_options=set(analysis_result.data_plane_machines.keys()),
        cli_input=cli_machines,
        parameter_type="Machine(s)",
    )
```